### PR TITLE
Set version to v0.29.1

### PR DESCRIPTION
**Pull request by albi3ro**
We will be making one final release of pennylane-sf in order to pin the version on pypi to `pennylane<v0.30.0`.

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,9 @@
-# Release 0.30.0-dev
-
-### New features since last release
+# Release 29.1
```
Comment (timmysilv) [inline .github/CHANGELOG.md:1]:
```suggestion
# Release 0.29.1
```

**Review by timmysilv — APPROVED** [review]
idk if it's a breaking change necessarily, but it doesn't really fit in any other category. lgtm! you can add yourself as a contributor if you want.. kinda looks like a ghost made this PR which is cool with me

**Review by trbromley — APPROVED** [review]
Thanks @albi3ro!

**Comment by trbromley** [comment]
> idk if it's a breaking change necessarily, but it doesn't really fit in any other category. lgtm! you can add yourself as a contributor if you want.. kinda looks like a ghost made this PR which is cool with me

@lillian542 also made a PR: https://github.com/PennyLaneAI/pennylane-sf/pull/134

**Pull request merged**